telemetry/definitions/commonDefinitions.json

```
@@ -773,13 +773,61 @@
             "name": "codecatalyst_devEnvironmentWorkflowStep",
             "type": "string",
             "description": "Workflow step name"
+        },
+        {
+            "name": "numberOfConnections",
+            "type": "int",
+            "description": "Number of Existing connections the user has."
+        },
+        {
+            "name": "builderIdStatus",
+            "type": "string",
+            "description": "login status with builderId option",
+            "allowedValues": [
+                "not_attempted",
+                "attempted",
+                "canceled", 
+                "failed"
+            ]
+        },
+        {
+            "name": "tab",
+            "type": "string",
+            "description": "Feature tab user is interacting in.",
+            "allowedValues": [
+                "awsResource",
+                "codewhisperer",
+                "codecatalyst"
+            ]
+        },
+        {
+            "name": "unlockedTabs",
+            "type": "string",
+            "description": "comma delimtied list of tabs that user already has valid connections to."
+        },
+        {
+            "name": "invalidInputFields",
+            "type": "string",
+            "description": "comma delimtied list of input fields that user has invalid inputs typed."
+
+        },
+        {
+            "name": "iamIdentityCenterRegion",
+            "type": "string",
+            "description": "Shows region selected by user under IAM Identity Center" 
+        },
+        {
+            "name": "userChoice",
+            "type": "string",
+            "description": "role switch choice user makes in quickpick when they add new profile" 
```

seems like this field could be used in more places than just this specific feature
```suggestion
            "description": "User selection from a predefined menu (not user-provided input)" 
```